### PR TITLE
Teachable Machine Post 작성

### PR DESCRIPTION
**Pull request by YubeenSon**
본 PR은 teachable machine post를 작성합니다.

### CODE REVIEW

File: _posts/2020-12-27-Teachable-Machine.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+tags: ["TeachableMachine"]
+image: assets/images/post-Teachable-Machine/teachable_machine.jpeg
+description: "Teachable Machine"
+featured: false
+---
+머신러닝이라는 것이 복잡하고 어려운 것만은 아니라고 저의 심리적 장벽을 낮춰준 것이 바로 Teachable Machine입니다. 이번 기회에 Teachable Machine을 공부한 내용을 기반으로 소개해보고자 합니다.  
+
+##  Teachable Machine  
+
+Teachable Mchine은 구글에서 제공하는 웹기반의 머신러닝 툴입니다. 여기서 머신 러닝 코드 없이 이미지, 소리, 포즈 자료를 업로드하여 인식 모델을 만들 수 있습니다. 기계 학습에 사전지식이 필요하지 않고, 툴의 이용이 쉽고 결과물을 빠르게 얻을 수 있다는 것이 특징입니다.
```
Comment (handal95) [inline _posts/2020-12-27-Teachable-Machine.md]:
```suggestion
Teachable Machine은 구글에서 제공하는 웹기반의 머신러닝 툴입니다. 여기서 머신 러닝 코드 없이 이미지, 소리, 포즈 자료를 업로드하여 인식 모델을 만들 수 있습니다. 기계 학습에 사전지식이 필요하지 않고, 툴의 이용이 쉽고 결과물을 빠르게 얻을 수 있다는 것이 특징입니다.   
```

File: _posts/2020-12-27-Teachable-Machine.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+title: "ML 맛보기 - Teachable Machine"
+authors: [YubeenSon]
+tags: ["TeachableMachine"]
+image: assets/images/post-Teachable-Machine/teachable_machine.jpeg
+description: "Teachable Machine"
+featured: false
+---
+머신러닝이라는 것이 복잡하고 어려운 것만은 아니라고 저의 심리적 장벽을 낮춰준 것이 바로 Teachable Machine입니다. 이번 기회에 Teachable Machine을 공부한 내용을 기반으로 소개해보고자 합니다.  
+
+##  Teachable Machine
```
Comment (handal95) [inline _posts/2020-12-27-Teachable-Machine.md]:
```suggestion
## Teachable Machine  
```

File: _posts/2020-12-27-Teachable-Machine.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+##  Teachable Machine  
+
+Teachable Mchine은 구글에서 제공하는 웹기반의 머신러닝 툴입니다. 여기서 머신 러닝 코드 없이 이미지, 소리, 포즈 자료를 업로드하여 인식 모델을 만들 수 있습니다. 기계 학습에 사전지식이 필요하지 않고, 툴의 이용이 쉽고 결과물을 빠르게 얻을 수 있다는 것이 특징입니다.   
+
+## 원리  
+
+Teachable Machine은 input받은 데이터를 미리 만들어진 신경망으로 전이학습을 시키는 것이기 때문에 복잡한 과정 없이 새로운 모델을 만들 수 있습니다. 다음은 teachable machine FAQ에 나오는 문장입니다.
+
+>There’s a pretrained neural network, and when you create your own classes, you can sort of picture that your classes are becoming the last layer or step of the neural net.” <sup>[1](#footnote_1)</sup>
```
Comment (handal95) [inline _posts/2020-12-27-Teachable-Machine.md]:
```suggestion
> ”There’s a pretrained neural network, and when you create your own classes, you can sort of picture that your classes are becoming the last layer or step of the neural net.” <sup>[1](#footnote_1)</sup>
```

File: _posts/2020-12-27-Teachable-Machine.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+(MobileNet : CNN에서 convolution을 depthwise separable convolution로 변경해 파라미터 수를 줄여 모바일 환경에서도 운영 가능하게 가벼운 모델)
+
+## 활용 방법(이미지 분류의 경우)
+
+![teachable_machine_page.jpeg](../assets/images/post-Teachable-Machine/teachable_machine_page.jpeg)
+출처 : 생활코딩
+
+1. 이미지를 업로드하거나, 웹캠을 통해 input 데이터를 만듭니다.
+2. 최소 2개의 클래스에 해당하는 input을 업로드 후 train model을 눌러 모델일 형성되기까지를 기다립니다.
```
Comment (handal95) [inline _posts/2020-12-27-Teachable-Machine.md]:
```suggestion
2. 최소 2개의 클래스에 해당하는 input을 업로드 후 train model을 눌러 모델이 형성되기까지를 기다립니다. 
```

**Review by handal95 — CHANGES_REQUESTED** [review]
일단 내용의 구성보다는 맞춤법들 위주로 확인했습니다.
워드파일이나 맞춤법 검사기 한 번 돌려보면 좋을 것 같아요.
머신 러닝 / 머신러닝 같은 용어들 통일하면 좋겠습니다

File: _posts/2020-12-27-Teachable-Machine.md (hunk 4 above)
Comment (YubeenSon) [reply _posts/2020-12-27-Teachable-Machine.md]: 자잘한 실수들 모두 반영하였습니다:3

File: _posts/2020-12-27-Teachable-Machine.md [hunk 5]
```diff
@@ -0,0 +1,53 @@
+---
+layout: post
+title: "ML 맛보기 - Teachable Machine"
+authors: [YubeenSon]
+tags: ["TeachableMachine"]
```
Comment (handal95) [inline _posts/2020-12-27-Teachable-Machine.md]: 태그에 머신러닝도 달려있으면 좋을 것 같아요

File: _posts/2020-12-27-Teachable-Machine.md [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+
+Teachable Machine은 input 받은 데이터를 미리 만들어진 신경망으로 전이학습을 시키는 것이기 때문에 복잡한 과정 없이 새로운 모델을 만들 수 있습니다. 다음은 Teachable Machine FAQ에 나오는 문장입니다.
+
+>"There’s a pretrained neural network, and when you create your own classes, you can sort of picture that your classes are becoming the last layer or step of the neural net.” <sup>[1](#footnote_1)</sup>
+
+(전이학습 : 사전에 훈련된 모델의 가중치를 가지고 와서 새로운 case에 재보정해서 사용하는 것)
+
+image, pose 프로젝트에는 MobileNet으로, audio 프로젝트에선 speech command recognizer<sup>[2](#footnote_2)</sup> 으로 미리 학습한 모델을 활용한다고 합니다. 
+
+(MobileNet : CNN에서 convolution을 depthwise separable convolution로 변경해 파라미터 수를 줄여 모바일 환경에서도 운영 가능하게 가벼운 모델)
```
Comment (handal95) [inline _posts/2020-12-27-Teachable-Machine.md]: MobileNet와 speech command recognizer 각주가 서로 양식이 다른데 
다른 각주스타일을 사용하신 의미가 있는건가요??

File: _posts/2020-12-27-Teachable-Machine.md [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+
+1. 이미지를 업로드하거나, 웹캠을 통해 input 데이터를 만듭니다.
+2. 최소 2개의 클래스에 해당하는 input을 업로드 후 train model을 눌러 모델이 형성되기까지를 기다립니다. 
+3. 모델이 만들어진 후 우측에서 성능을 테스트합니다. 
+
+### Advanced
+ 
+가운데 training엔 advanced라고 모델 형성 시 몇몇 부분을 조정할 수 있게 만들어두었습니다. 조정 가능한 항목은 epoch, batch size, learning rate입니다.  
+
+Epoch는 전체 데이터를 몇 번 사용해서 학습을 시킬지를 뜻합니다. epoch가 너무 커도 좋지 않은 것이 epoch가 높으면 과적합(overfitting)되어 새로운 데이터에 대한 예측률이 떨어질 수 있습니다.
```
Comment (handal95) [inline _posts/2020-12-27-Teachable-Machine.md]: overfitting문제때문에 epoch를 너무 크게 잡으면 안된다면, 항상 작게 설정하면 좋은건가요?

File: _posts/2020-12-27-Teachable-Machine.md [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+
+Batch Size 한 번의 batch마다 주는 데이터 샘플의 크기를 뜻합니다. 여기서 batch는 나눠진 데이터 셋을 의미합니다. 메모리의 한계와 속도 저하의 문제로 모든 데이터를 한 번에 집어넣을 수 없어서 batch 단위로 데이터를 투입하게 됩니다. 주로 16, 32, 64를 사용합니다.
+
+Learning rate는 최적값을 찾아가는 속도로 보통 0.01로 맞춰 놓습니다. 학습 후 결과가 발산(overshooting)하면 작은 값을, 학습하는데 시간이 너무 오래 걸리면 큰 값으로 수정하면 됩니다.
```
Comment (handal95) [inline _posts/2020-12-27-Teachable-Machine.md]: 결과가 발산하면 작은 값을, 시간이 너무 오래걸리면 큰 값으로 수정하는 이유가 있나요??
learning rate에 대한 설명이 조금 더 있으면 좋을 것 같습니다.

**Review by handal95 — CHANGES_REQUESTED** [review]
이번에는 내용상으로 추가하면 좋을 것 같은 부분들을 체크해봤습니다!
밤늦게 고생 많으십니다 👍 👍

File: _posts/2020-12-27-Teachable-Machine.md (hunk 6 above)
Comment (YubeenSon) [reply _posts/2020-12-27-Teachable-Machine.md]: speech command recognizer의 경우  teachable machine 구축을 위해 개발진들이 만든 것 같아서 설명 대신 별도의 각주로 처리했습니다. 통일성을 위해 MobileNet도 설명각주 추가하였습니다!

File: _posts/2020-12-27-Teachable-Machine.md (hunk 7 above)
Comment (handal95) [reply _posts/2020-12-27-Teachable-Machine.md]:
```suggestion
Epoch : 전체 데이터 셋에 대해 한 번 학습을 완료한 상태(epochs = 40이라면 전체 데이터를 40번 사용해서 학습을 거치는 것) 
    - Epoch가 너무 작다면, 학습횟수가 부족해서 적절한 학습을 달성하지 못합니다.
    - Epoch가 너무 크다면, 기존 데이터 셋에 대한 의존도가 너무 큰 과적합(overfitting)이 발생해서, 새로운 데이터에 대한 예측률이 떨어질 수 있습니다.
```
내용을 이렇게 보충하는건 어떨까요

File: _posts/2020-12-27-Teachable-Machine.md (hunk 8 above)
Comment (handal95) [reply _posts/2020-12-27-Teachable-Machine.md]: Batch Size : 메모리의 한계와 속도 저하문제로 모든 데이터를 한번에 집어넣을 수 없으므로, 학습에 사용하기 위해 분할하는 데이터(batch) 샘플들의 크기.
    - 총 데이터가 512개, batch size가 16이라면, 1번의 epoch(512개의 데이터) 당 batch size(16개의 데이터)단위로 32번의 반복(iterator)학습을 진행합니다.

```suggestion
Learning rate는 현재 학습의 결과를 다음 번 학습에 반영하는 효율. 일반적으로 0.01의 값을 시작값으로 사용해서, 상황에 따라 적절히 수정해서 사용합니다.
    - Learning rate이 너무 작다면, 목표 최적값에 수렴하기 위한이 굉장히 많은 시간이 소요됩니다.
    - Learning rate이 너무 크다면, 학습 반영이 과다하게 이루어져서 목표값에 수렴하지 못하고 발산(overshooting)하게 됩니다.
```
gradient descent 같은 설명이 없어서(추가하게 되면 내용이 Teachable Machine에서 너무 벗어난 ML소개가 될까봐) 조금 추상적으로 적었더니 용어 설명이 엄밀하지는 않네요! 그래도 이렇게 수정하면, 관련 포스트 처음 보는 분들도 용어에 대해서 어느정도는 겉핥기가 되지 않을까 싶습니다!

**Comment by handal95** [comment]
새벽을 틈타 author 추가 탑승합니다 ㅎㅎ

**Comment by YubeenSon** [comment]
다 되었습니다~~

**Comment by handal95** [comment]
> 다 되었습니다~~

고생...해씁니다 화이팅!! 
떠나보냅시당 🚀

**Review by handal95 — APPROVED** [review]
